tests/generators/kzg_4844/main.py

```diff
@@ -0,0 +1,580 @@
+        },
+        'output': None
+    }
+
+    # Edge case: Invalid proof, not on curve
+    blob = VALID_BLOBS[1]
+    proof = P1_NOT_ON_CURVE
+    commitment = bls.G1_to_bytes48(bls.G1())
+    expect_exception(spec.verify_blob_kzg_proof, blob, commitment, proof)
+    yield 'verify_blob_kzg_proof_case_proof_not_on_curve', {
+        'input': {
+            'blob': encode_hex(blob),
+            'commitment': encode_hex(commitment),
+            'proof': encode_hex(proof),
+        },
+        'output': None
+    }
+
+    # Edge case: Invalid commitment, not in G1
+    blob = VALID_BLOBS[0]
+    proof = bls.G1_to_bytes48(bls.G1())
+    commitment = P1_NOT_IN_G1
+    expect_exception(spec.verify_blob_kzg_proof, blob, commitment, proof)
+    yield 'verify_blob_kzg_proof_case_commitment_not_in_G1', {
+        'input': {
+            'blob': encode_hex(blob),
+            'commitment': encode_hex(commitment),
+            'proof': encode_hex(proof),
+        },
+        'output': None
+    }
+
+    # Edge case: Invalid commitment, not on curve
+    blob = VALID_BLOBS[2]
+    proof = bls.G1_to_bytes48(bls.G1())
+    commitment = P1_NOT_ON_CURVE
+    expect_exception(spec.verify_blob_kzg_proof, blob, commitment, proof)
+    yield 'verify_blob_kzg_proof_case_commitment_not_on_curve', {
+        'input': {
+            'blob': encode_hex(blob),
+            'commitment': encode_hex(commitment),
+            'proof': encode_hex(proof),
+        },
+        'output': None
+    }
+
+    # Edge case: Invalid blob
+    for blob in INVALID_BLOBS:
+        proof = bls.G1_to_bytes48(bls.G1())
+        commitment = bls.G1_to_bytes48(bls.G1())
+        expect_exception(spec.verify_blob_kzg_proof, blob, commitment, proof)
+        identifier = f'{encode_hex(hash(blob))}'
+        yield f'verify_blob_kzg_proof_case_invalid_blob_{(hash(bytes(identifier, "utf-8"))[:8]).hex()}', {
+            'input': {
+                'blob': encode_hex(blob),
+                'commitment': encode_hex(commitment),
+                'proof': encode_hex(proof),
+            },
+            'output': None
+        }
+
+
+def case06_verify_blob_kzg_proof_batch():
+    # Valid cases
+    proofs = []
+    commitments = []
+    for blob in VALID_BLOBS:
+        proofs.append(spec.compute_blob_kzg_proof(blob))
+        commitments.append(spec.blob_to_kzg_commitment(blob))
+
+    for i in range(len(proofs)):
+        assert spec.verify_blob_kzg_proof_batch(VALID_BLOBS[:i], commitments[:i], proofs[:i])
+        identifier = f'{encode_hex(hash(b"".join(VALID_BLOBS[:i])))}'
+        yield f'verify_blob_kzg_proof_batch_case_{(hash(bytes(identifier, "utf-8"))[:8]).hex()}', {
+            'input': {
+                'blobs': encode_hex_list(VALID_BLOBS[:i]),
+                'commitments': encode_hex_list(commitments[:i]),
+                'proofs': encode_hex_list(proofs[:i]),
+            },
+            'output': True
+        }
+
+    # Incorrect proof
```

Comment:
Hmm, do we also need a test vector where batch is called with no blobs/commitments?

Reply:
This is included. `for i in range(len(proofs)):` includes `i == 0`.

Reply:
ACK. Missed that!